R/FilterStateChoices.R

```diff
@@ -201,7 +201,12 @@ ChoicesFilterState <- R6::R6Class( # nolint
       do.call(super$initialize, args)
 
       private$set_choices(choices)
-      private$set_selected(selected)
+
+      if (is.null(selected) && isFALSE(multiple)) {
```

Comment:
at this stage `multiple` cannot be `NULL`

I wonder, should this check be in the constructor or in `$set_choices`?

Reply:
My mistake, I misread as `is.null(multiple)`.

Reply:
Should it then not be like this?
```
      if (is.null(selected)) {
        if (multiple) {
          private$set_selected(choices)
        } else {
          private$set_selected(choices[1])
        }
      } else {
        private$set_selected(selected)
      }
```

Reply:
I have updated code to fix https://github.com/insightsengineering/teal.slice/pull/296#discussion_r1210319410 and optimized this logic as well.